src/main/kotlin/org/jetbrains/research/testspark/helpers/psi/java/JavaPsiHelper.kt

```diff
@@ -0,0 +1,205 @@
+import com.intellij.openapi.diagnostic.Logger
+import com.intellij.openapi.editor.Caret
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.util.TextRange
+import com.intellij.psi.PsiClass
+import com.intellij.psi.PsiDocumentManager
+import com.intellij.psi.PsiElement
+import com.intellij.psi.PsiFile
+import com.intellij.psi.PsiMethod
+import com.intellij.psi.util.PsiTreeUtil
+import com.intellij.psi.util.PsiTypesUtil
+import org.jetbrains.research.testspark.helpers.psi.Language
+import org.jetbrains.research.testspark.helpers.psi.PsiClassWrapper
+import org.jetbrains.research.testspark.helpers.psi.PsiHelper
+import org.jetbrains.research.testspark.helpers.psi.PsiMethodWrapper
+import org.jetbrains.research.testspark.tools.llm.SettingsArguments
+
+class JavaPsiHelper(private val psiFile: PsiFile) : PsiHelper {
+
+    override val language: Language get() = Language.Java
+
+    private val log = Logger.getInstance(this::class.java)
+
+    override fun generateMethodDescriptor(
+        psiMethod: PsiMethodWrapper,
+    ): String {
+        val methodDescriptor = psiMethod.methodDescriptor
+        log.info("Method description: $methodDescriptor")
+        return methodDescriptor
+    }
+
+    override fun getSurroundingClass(
+        caretOffset: Int,
+    ): PsiClassWrapper? {
+        val classElements = PsiTreeUtil.findChildrenOfAnyType(psiFile, PsiClass::class.java)
+        for (cls in classElements) {
+            if (cls.containsOffset(caretOffset)) {
+                val javaClassWrapper = JavaPsiClassWrapper(cls)
+                if (javaClassWrapper.isTestableClass()) {
+                    log.info("Surrounding class for caret in $caretOffset is ${javaClassWrapper.qualifiedName}")
+                    return javaClassWrapper
+                }
+            }
+        }
+        log.info("No surrounding class for caret in $caretOffset")
+        return null
+    }
+
+    override fun getSurroundingMethod(
+        caretOffset: Int,
+    ): PsiMethodWrapper? {
+        val methodElements = PsiTreeUtil.findChildrenOfAnyType(psiFile, PsiMethod::class.java)
+        for (method in methodElements) {
+            if (method.body != null && method.containsOffset(caretOffset)) {
+                val surroundingClass =
+                    PsiTreeUtil.getParentOfType(method, PsiClass::class.java) ?: continue
+                val surroundingClassWrapper = JavaPsiClassWrapper(surroundingClass)
+                if (surroundingClassWrapper.isTestableClass()) {
+                    val javaMethod = JavaPsiMethodWrapper(method)
+                    log.info("Surrounding method for caret in $caretOffset is ${javaMethod.methodDescriptor}")
+                    return javaMethod
+                }
+            }
+        }
+        log.info("No surrounding method for caret in $caretOffset")
+        return null
+    }
+
+    override fun getSurroundingLine(
+        caretOffset: Int,
+    ): Int? {
```

Comment:
I think the name `getSurroundingLineNumber` (or `getSurroundingLineIndex` if the index is returned) is better; as for now, the name suggests that the method returns the actual line as `String`.

Reply:
https://github.com/JetBrains-Research/TestSpark/issues/235#issue-2331296061